Src/spinlex.c

```diff
@@ -1575,15 +1611,43 @@ lex(void)
 	}
 
 	if (isdigit_(c))
-	{	long int nr;
-		getword(c, isdigit_);
-		errno = 0;
-		nr = strtol(yytext, NULL, 10);
-		if (errno != 0)
-		{	fprintf(stderr, "spin: value out of range: '%s' read as '%d'\n",
-				yytext, (int) nr);
+	{	getword(c, isdigit_);
```

Comment:
Is this feasible to refactor this piece of code? This is an algorithm that we add, not just adapt, so I think we could try to follow good programming practices, if possible.
Can this code (or its parts) be extracted to reduce nesting and cyclomatic complexity?
Can we avoid reassining new values to existing variables - this is dangerous, especially as we have consecutive ifs without returns which rely on a value that is changed inside.